### PR TITLE
[Enhance] Support classwise evaluation in CocoPanopticDataset

### PR DESCRIPTION
**Pull request by AronLin**
Support classwise evaluation in CocoPanopticDataset.

### CODE REVIEW

**Comment by AronLin** [comment]
Should I add a unit test for CocoPanopticDataset in this PR? @ZwwWayne

**Comment by ZwwWayne** [comment]
> Should I add a unit test for CocoPanopticDataset in this PR? @ZwwWayne 

we should try that. Furthermore, may post a picture here to show what it looks like.

File: tests/test_data/test_datasets/test_panoptic_dataset.py [hunk 1]
```diff
@@ -141,3 +147,151 @@ def test_load_panoptic_style_json():
     assert ann['bboxes'].shape[0] == ann['labels'].shape[0] == 1
     assert ann['bboxes_ignore'].shape[0] == 1
     assert len(ann['masks']) == 3
+
+
+def test_panoptic_evaluation():
+    if id2rgb is None:
```
Comment (ZwwWayne) [inline tests/test_data/test_datasets/test_panoptic_dataset.py:153]: Extract a function named _create_panoptic_gt_pred()

File: tests/test_data/test_datasets/test_panoptic_dataset.py [hunk 2]
```diff
@@ -141,3 +147,161 @@ def test_load_panoptic_style_json():
...
+        'id': 2,
+        'name': 'wall',
+        'supercategory': 'wall',
+        'isthing': 0
+    }]
+
+    images = [{
+        'id': 0,
+        'width': 800,
+        'height': 600,
```
Comment (RangiLyu) [inline tests/test_data/test_datasets/test_panoptic_dataset.py:173]: 60, 80

File: tests/test_data/test_datasets/test_panoptic_dataset.py [hunk 3]
```diff
@@ -141,3 +147,161 @@ def test_load_panoptic_style_json():
...
+        [
+            [11, 11, 10, 40],  # TP IoU=351/449=0.78
+            [38, 10, 10, 40],  # FP
+            [51, 10, 10, 5]
+        ],  # TP IoU=45/55=0.818
+        dtype=np.int64)
+    pred_labels = np.array([0, 0, 1], dtype=np.int64)
+    for i in range(3):
+        x, y, w, h = pred_bboxes[i]
+        pred[y:y + h, x:x + w] = (i + 1) * 1000 + pred_labels[i]
```
Comment (RangiLyu) [inline tests/test_data/test_datasets/test_panoptic_dataset.py:248]: use instance offset in dataset

**Review by RangiLyu — APPROVED** [review]
LGTM

**Pull request merged**